src/ucp/wireup/address.c

```diff
@@ -597,8 +597,11 @@ static ucs_status_t ucp_address_do_pack(ucp_worker_h worker, ucp_ep_h ep,
                                       context->tl_rscs[rsc_index].tl_name_csum);
 
             /* Transport information */
-            enable_amo = (flags & UCP_ADDRESS_PACK_FLAG_HW_AMO_TLS) &&
-                         worker->atomic_tls & UCS_BIT(rsc_index);
+            enable_amo = (flags & UCP_ADDRESS_PACK_FLAG_ENABLE_DEVICE_ATOMICS) &&
```

Comment:
IMHO, it's maybe simpler in implementation but more confusing from API and naming perspective. CPU atomics also may be kind of HW atomics

Reply:
CPU atomics are not "kind of" HW atomics: each iface export a flag which says whether it's CPU atomics or device atomics. atomic_tls will be the same as today. cpu_atomic_tls will be the tls which export UCT_IFACE_FLAG_ATOMIC_CPU

Reply:
Ok, if I disable UCP_ADDRESS_PACK_FLAG_HW_AMO_TLS flag, should be AMO capabilities be packed into shared memory iface address (no matter on which bitmap it's saved inside worker structure)? As I understand it's HW AMO, is not it?

Reply:
what's the point to have magic "worker->atomic_tls" if we can have cpu and device bitmaps separately? this is the only place where some extra logic is needed and it's not seems to be too complicated because meaning of every variable and flag is clear, conditions are ordered by priority, that's it.